### PR TITLE
feat: add hiddenLabel prop to Checkbox component

### LINKED ISSUE #203: Visually hidden `Checkbox` labels
**Is your feature request related to a problem? Please describe.**
For accessibility, we need a way to render a `Checkbox` component without a visual label. For instance, a selectable `Table.Row` needs a `Checkbox`, but with a visually-hidden label that specifies the actions to be made.

**Describe the solution, feature, or improvement you'd like**
Open to recommendations but here are some possible solutions:

Add a `visuallyHidden` prop to the `Checkbox.Label` element and add conditional visually-hidden styles. This will allow someone have an implementation like so:
```tsx
<Checkbox label={<Checkbox.Label visuallyHidden>...</Checkbox.Label>} />
```

Another possible solution is to just add a prop to the `Checkbox` component:
```tsx
<Checkbox label="..." visuallyHidden />
```

**Additional context**
We changed the `label` prop to be optional during the meantime, but we'll need to make it required again with this improvement.

### PR DESCRIPTION
**Pull request by jordan-massingill**
#WHAT 
Adds `hiddenLabel` prop to Checkbox component to allow for visually hidden label, maintain accessibility. Closes #203.

#TESTING 
Tested locally to ensure prop allows component to render correctly

### CODE REVIEW

File: packages/big-design/src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { uniqueId } from '../../utils';
 import { CheckboxContainer, HiddenCheckbox, StyledCheckbox, StyledLabel } from './styled';
 
 interface Props {
+  hiddenLabel?: boolean;
   isIndeterminate?: boolean;
   label?: React.ReactChild;
```
Comment (chanceaclark) [inline packages/big-design/src/components/Checkbox/Checkbox.tsx]: We probably want to add back `label` as required again. See #201 on why we made it optional. Afterwards, you may want to go into the table components and update the Checkboxes accordingly.

Comment (jordan-massingill) [reply packages/big-design/src/components/Checkbox/Checkbox.tsx]: Yeah, I went back and forth on whether to do that or make it part of another ticket. If this all looks okay, I'll go ahead and update the table component too.

File: packages/big-design/src/components/Table/Row/Row.tsx [hunk 2]
```diff
@@ -28,18 +30,20 @@ const InternalRow = <T extends TableItem>({
     }
   };
 
+  const label = isSelected ? `Deslect item number ${itemIndex + 1}` : `Select item number ${itemIndex + 1}`;
```
Comment (chanceaclark) [inline packages/big-design/src/components/Table/Row/Row.tsx]: Typo:
```suggestion
  const label = isSelected ? `Deselect item number ${itemIndex + 1}` : `Select item number ${itemIndex + 1}`;
```

🍹What about renaming to something like "Deselect row {number}"?

File: packages/big-design/src/components/Table/SelectAll/SelectAll.tsx [hunk 3]
```diff
@@ -51,7 +51,13 @@ export const SelectAll = <T extends TableItem>({
   return (
     <Flex.Item marginRight="xxSmall">
       <Flex flexDirection="row">
-        <Checkbox isIndeterminate={someInPageSelected} checked={allInPageSelected} onChange={handleSelectAll} />
+        <Checkbox
+          isIndeterminate={someInPageSelected}
+          hiddenLabel
+          label="Select All"
```
Comment (chanceaclark) [inline packages/big-design/src/components/Table/SelectAll/SelectAll.tsx]: Might want to add some logic around deselecting all rows as well.

File: packages/big-design/src/components/Table/Row/Row.tsx (hunk 2 above)
Comment (deini) [reply packages/big-design/src/components/Table/Row/Row.tsx]: Do we need the item index? The Row already has the context of the `item`, is there any benefits of having the index as part of the label?

Comment (jordan-massingill) [reply packages/big-design/src/components/Table/Row/Row.tsx]: This was a suggestion from @chanceaclark as an option for an accessible label, but the label for the checkbox was arbitrarily chosen and can be changed. Let's discuss offline 🌮

**Pull request merged**